### DISCUSSION TITLE
Suggestion: Time Duration property editor

### DISCUSSION THREAD

**Discussion by hfloyd** [leekelleher/umbraco-contentment#114] [Idea]
If you are looking for an  idea for another property editor...
How about one for time duration - like years/months/days/hours/mins/seconds?

I was looking for one today, and only came across this forum thread: https://our.umbraco.com/forum/umbraco-8/99027-timepicker-property-editor

I envision the config to allow the selection of the max-min options (ex: hours-seconds, months-hours, years-days, etc.) so an appropriate range can be provided (like, for the duration of a class, perhaps days-minutes would be appropriate options, task milestones might be months-days...)
![image](https://user-images.githubusercontent.com/1252593/119929071-ce9fdb00-bf4a-11eb-9b1b-1aa333148eef.png)

The UI for the Property editor would show separate number boxes for each available option:

![image](https://user-images.githubusercontent.com/1252593/119928536-c2674e00-bf49-11eb-936c-efbd5b92da0f.png)

And a PropertyValueEditor could return the data as a Timespan struct...

**Comment by leekelleher** [comment]
@hfloyd I like this idea! 🎉 

Several times (over the years) I've had a need to enter a duration - for videos, courses, recipe timings, etc. I'd typically ended up using a numeric value of the lowest dominator, e.g. number of seconds/minutes - then calculating the `TimeSpan` from that value.

I've give some thought on how it could be implemented. 💡

**Reply by hfloyd** [reply]
Yay! Yes, I've done that before as well - picking something like seconds or minutes... but not as editor-friendly. (Seems you always have to get out a calculator...) Also, just looking at the number of seconds in a textbox doesn't give you a good ideas of the value at a glance...

**Reply by hfloyd** [reply]
Ps. I suck at Angular, but am happy to help in whatever way, even if just as a tester ;-)

**Comment by hfloyd** [comment]
Looking closer at the [TimeSpan struct](https://docs.microsoft.com/en-us/dotnet/api/system.timespan?view=netframework-4.7.2#remarks), I see that "days" is the maximum aspect, which makes sense, since "month" and "year" are variable, based on the specific month/year in question, and the idea of "Duration" is a quantity separate from specific DateTimes. Knowing a number of "days" one could calculate months/years if a specific "start" date was provided.

So, I'd like to amend my request to only allow options from days down to seconds, to match the TimeSpan:
![image](https://user-images.githubusercontent.com/1252593/121099091-3533bd00-c7c5-11eb-800b-33e1f35e37ec.png)

**Comment by hfloyd** [comment]
FYI - I came across this: https://angular-slider.github.io/angularjs-slider/ 
Not sure if it might help?

**Comment by leekelleher** [comment]
Closing this off, as I haven't needed to re-explore the duration idea. Happy to reopen this if anyone wants to revisit.
